evm

```diff
@@ -118,23 +118,79 @@ function evm_system() {
   export PATH
 }
 
-function _unzip-strip() {
+function _unzip_strip() {
   # shopt is a bash built-in command. Zsh, has setopt instead
   # so, set the command according to the type of shell (only supports bash and zsh)
   local SHOPT="shopt"
-  if [[ "$EVM_PS_SHELL" == *zsh ]]; then
+  if [[ "$EVM_PS_SHELL" == *zsh ]] || [[  "$(which $SHELL)" == *"/zsh" ]] ; then
     SHOPT="setopt"
   fi
 
   local zip="$1"
   local dest="${2:-.}"
-  local temp=$(mktemp -d) && unzip -d "$temp" "$zip" && mkdir -p "$dest" && $SHOPT -s dotglob && local f=("$temp"/*) &&
+  local -i _err=0
+  local temp=$(mktemp -d)
+  _err=$?
+  if [ ${_err} -ne 0 ] ; then
+  {
+    return ${_err}
+  }
+  fi
+  unzip -d "$temp" "$zip"
+  _err=$?
+  if [ ${_err} -ne 0 ] ; then
+  {
+    return ${_err}
+  }
+  fi
```

Comment:
I understand what you are trying to do here and why, but you have this piece of code repeated at least 8 times! What can you do to make it better? Other parts of the code clearly do not handle errors like this, so we should be consistent.

Reply:
This code might look repeated, but there is not other way around it, $err can only be catch-ed after the execution, and I asking for the function to fail with error number and do not move forward, otherwise it gets very messy. Not a fan on letting errors explode and having to decode and clean a long list of errors after this fails since this step is needed for the next to happen.

Reply:
If the goal here is to just abort the execution on any error and return the same error code, then the best option would be just to use `set -o errexit` at the beginning of the script, which would "exit upon error, avoiding cascading errors", no?

Reply:
set -o errexit   this I have not used yet. But I was not going to fail on every error, just to exist in this logic here. Feel free to remove them all if you want